mldsa/mldsa_native.h

```diff
@@ -0,0 +1,564 @@
+#error MLD_CONFIG_API_PARAMETER_SET not defined
+#endif
+#if !defined(MLD_CONFIG_API_NAMESPACE_PREFIX)
+#error MLD_CONFIG_API_NAMESPACE_PREFIX not defined
+#endif
+
+/* Validate parameter set */
+#if MLD_CONFIG_API_PARAMETER_SET != 44 && \
+    MLD_CONFIG_API_PARAMETER_SET != 65 && MLD_CONFIG_API_PARAMETER_SET != 87
+#error MLD_CONFIG_API_PARAMETER_SET must be 44, 65, or 87
+#endif
+
+/* Derive namespacing macro */
+#define MLD_API_CONCAT_(x, y) x##y
+#define MLD_API_CONCAT(x, y) MLD_API_CONCAT_(x, y)
+#define MLD_API_CONCAT_UNDERSCORE(x, y) MLD_API_CONCAT(MLD_API_CONCAT(x, _), y)
+#define MLD_API_NAMESPACE(sym) \
+  MLD_API_CONCAT_UNDERSCORE(MLD_CONFIG_API_NAMESPACE_PREFIX, sym)
+
+#if defined(__GNUC__) || defined(clang)
+#define MLD_API_MUST_CHECK_RETURN_VALUE __attribute__((warn_unused_result))
+#else
+#define MLD_API_MUST_CHECK_RETURN_VALUE
+#endif
+
+#include <stddef.h>
+#include <stdint.h>
+
+/*************************************************
+ * Name:        crypto_sign_keypair_internal
+ *
+ * Description: Generates public and private key. Internal API.
+ *              When MLD_CONFIG_KEYGEN_PCT is set, performs a Pairwise
+ *              Consistency Test (PCT) as required by FIPS 140-3 IG.
+ *
+ * Arguments:   - uint8_t *pk:   pointer to output public key (allocated
+ *                               array of MLDSA_PUBLICKEYBYTES bytes)
+ *              - uint8_t *sk:   pointer to output private key (allocated
+ *                               array of MLDSA_SECRETKEYBYTES bytes)
+ *              - uint8_t *seed: pointer to input random seed (MLDSA_SEEDBYTES
+ *                               bytes)
+ *
+ * Returns 0 (success) or -1 (PCT failure)
+ *
+ * Specification: Implements @[FIPS204 Algorithm 6 (ML-DSA.KeyGen_internal)]
+ *
+ **************************************************/
+MLD_API_MUST_CHECK_RETURN_VALUE
+int MLD_API_NAMESPACE(keypair_internal)(
+    uint8_t pk[MLDSA_PUBLICKEYBYTES(MLD_CONFIG_API_PARAMETER_SET)],
+    uint8_t sk[MLDSA_SECRETKEYBYTES(MLD_CONFIG_API_PARAMETER_SET)],
+    const uint8_t seed[MLDSA_SEEDBYTES]);
+
+/*************************************************
+ * Name:        crypto_sign_keypair
+ *
+ * Description: Generates public and private key.
+ *              When MLD_CONFIG_KEYGEN_PCT is set, performs a Pairwise
+ *              Consistency Test (PCT) as required by FIPS 140-3 IG.
+ *
+ * Arguments:   - uint8_t *pk:   pointer to output public key (allocated
+ *                               array of MLDSA_PUBLICKEYBYTES bytes)
+ *              - uint8_t *sk:   pointer to output private key (allocated
+ *                               array of MLDSA_SECRETKEYBYTES bytes)
+ *
+ * Returns 0 (success) or -1 (PCT failure)
+ *
+ * Specification: Implements @[FIPS204 Algorithm 1 (ML-DSA.KeyGen)]
+ *
+ **************************************************/
+MLD_API_MUST_CHECK_RETURN_VALUE
+int MLD_API_NAMESPACE(keypair)(
+    uint8_t pk[MLDSA_PUBLICKEYBYTES(MLD_CONFIG_API_PARAMETER_SET)],
```

Comment:
There are some minor inconsistencies between the API in `mldsa_native.h` and `sign.h`: Here we use array size indicators, while in `sign.h.`, we work with raw pointers. That should be uniform as in mlkem-native.